.gitignore

```diff
@@ -1 +1,7 @@
 CMakeLists.txt.user*
+
+# Build artifacts
+build/
```

Comment:
No, don't add this. Builds should not happen inside the source tree.

Reply:
The repository's ReadMe suggests doing just that. I've seen several other projects follow the same format.

What's so wrong with adding the build/ directory to this file?

Reply:
At the very least, if it's going to be added to the repository, it needs to happen in it's own pull request.

There are a variety of reasons why builds should not be done inside the source directory, but one specific one that I'll share to help you research the topic, is that cmake natively understands multiple configurations. If you put the build inside the source directory, you pre-clude the ability to have multiple build configurations at the same time.

Reply:
I can remove it. It's just that it was making my life easier as I've always built libraries this way.